enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,249 @@
+
+### Workflow Description
+
+1. Administrator edits MicroShift config file to specify desired audit logging policy profile
+2. Administrator edits MicroShift config file to specify max file size, number of files total, and max age of files for logs
+3. MicroShift service is restarted to apply changes
+
+### API Extensions
+
+**Audit Log Policy Configuration:**
+
+- MicroShift config’s `apiServer` root-field must include a child node for storing log configurations.
+
+```yaml
+apiServer:
+  auditLog: MAP(STRING)INTERFACE
+```
+
+- The `apiServer.auditLog` field must include a child node for storing the audit log policy profile to use. See [Implementation Details](#implementation-detailsnotesconstraints) for more. These are the same policies are equal to [Openshift Audit log policies](https://docs.openshift.com/container-platform/4.14/security/audit-log-policy-config.html#about-audit-log-profiles_audit-log-policy-config).
+
+```yaml
+apiServer:
+  auditLog:
+    policy: STRING
+```
+
+**Audit Log File Rotation:**
+
+Log rotation is governed by 3 values defining file size, number of retained files, and the maximum age of retained files.  The values are processed independently of one another, without prioritization.  Max size and max number of files may be used in conjunction to limit the storage footprint of retained logs (e.g. `maxFileSize` * `maxFiles` = log storage upper limit)
+
+- `maxFileSize` specifies maximum audit log file size. Accepted values must be a string concatenation of an integer and unit suffix with no separator character, e.g. `10GB`. Valid units are "MB", "GB", "TB". Since the kube-apiserver only accepts values defined in whole megabytes, MicroShift will need to translate this field value internally. This is trivial to implement and provides a bit of convenience to users.
+- `maxFiles` specifies the maximum number of rotated audit log files to retain.  Once this limit is reached, the apiserver will delete log files in order from oldest to newest, until all specified limits are satisfied. 
+  - _For example_,`maxFiles: 1` results will result in only 1 rotated file being retained, provided it also is within the `maxFileAge` limit, if specified.
+- `maxFileAge` specifies the maximum length of time to retain log files.  Files older than this limit will be deleted.
```

Comment:
Please, explain the format of `maxFileAge`.

Reply:
Woops, good catch.  Will fix